include/boost/math/quadrature/detail/tanh_sinh_detail.hpp

```diff
@@ -0,0 +1,689 @@
+// Copyright Nick Thompson, 2017
+// Use, modification and distribution are subject to the
+// Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_MATH_QUADRATURE_DETAIL_TANH_SINH_DETAIL_HPP
+#define BOOST_MATH_QUADRATURE_DETAIL_TANH_SINH_DETAIL_HPP
+
+#include <cmath>
+#include <vector>
+#include <boost/math/tools/atomic.hpp>
+#include <boost/detail/lightweight_mutex.hpp>
+#include <typeinfo>
+#include <boost/math/constants/constants.hpp>
+#include <boost/math/special_functions/next.hpp>
+
+
```

Comment:
I keep trying to find a way to make this work, with GCC-6.3 and:

#ifdef __GNUC__
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wliteral-range"
#pragma GCC diagnostic ignored "-Woverflow"
#endif

I still see:

In file included from ..\..\../boost/math/quadrature/tanh_sinh.hpp:35:0,
                 from tanh_sinh_quadrature_test.cpp:17:
..\..\../boost/math/quadrature/detail/tanh_sinh_detail.hpp:561:7: warning: floating constant truncated to zero [-Woverflow]

Yet strangely, -Wno-overflow does silence things when use don the command line :(

Using #pragama GCC system_header seems to work though - not ideal but it will have to do I guess.